railties/lib/rails/tasks/restart.rake

```diff
@@ -1,4 +1,5 @@
 desc "Restart app by touching tmp/restart.txt"
 task :restart do
+  Dir.mkdir('tmp') unless File.directory('tmp')
```

Comment:
Can you add a test for this? :)

https://github.com/rails/rails/blob/master/railties/test/application/rake/restart_test.rb